packages/module/src/providers/blockfrost.provider.ts

```diff
@@ -114,76 +88,50 @@ export class BlockfrostProvider implements IFetcher, ISubmitter {
   }
 
   async fetchProtocolParameters(epoch = Number.NaN): Promise<Protocol> {
-    try {
-      const { data, status } = await this._axiosInstance.get(
-        `epochs/${isNaN(epoch) ? 'latest' : epoch}/parameters`
-      );
-
-      if (status === 200)
-        return <Protocol>{
-          coinsPerUTxOSize: data.coins_per_utxo_word,
-          collateralPercent: data.collateral_percent,
-          decentralisation: data.decentralisation_param,
-          epoch: data.epoch,
-          keyDeposit: data.key_deposit,
-          maxBlockExMem: data.max_block_ex_mem,
-          maxBlockExSteps: data.max_block_ex_steps,
-          maxBlockHeaderSize: data.max_block_header_size,
-          maxBlockSize: data.max_block_size,
-          maxCollateralInputs: data.max_collateral_inputs,
-          maxTxExMem: data.max_tx_ex_mem,
-          maxTxExSteps: data.max_tx_ex_steps,
-          maxTxSize: data.max_tx_size,
-          maxValSize: data.max_val_size,
-          minFeeA: data.min_fee_a,
-          minFeeB: data.min_fee_b,
-          minPoolCost: data.min_pool_cost,
-          poolDeposit: data.pool_deposit,
-          priceMem: data.price_mem,
-          priceStep: data.price_step,
-        };
-
-      throw parseHttpError(data);
-    } catch (error) {
-      throw parseHttpError(error);
-    }
+    const latestEpochData = await this.blockfrostAPI.epochsLatest();
+    const epochNumber = isNaN(epoch) ? latestEpochData.epoch : epoch;
+    const bfResponse = await this.blockfrostAPI.epochsParameters(epochNumber);
+
+    return {
+      coinsPerUTxOSize: bfResponse.coins_per_utxo_word || '',
+      collateralPercent: bfResponse.collateral_percent || 0,
+      decentralisation: bfResponse.decentralisation_param,
+      epoch: bfResponse.epoch,
+      keyDeposit: bfResponse.key_deposit,
+      maxBlockExMem: bfResponse.max_block_ex_mem || '',
+      maxBlockExSteps: bfResponse.max_block_ex_steps || '',
+      maxBlockHeaderSize: bfResponse.max_block_header_size,
+      maxBlockSize: bfResponse.max_block_size,
+      maxCollateralInputs: bfResponse.max_collateral_inputs || 0,
+      maxTxExMem: bfResponse.max_tx_ex_mem || '',
+      maxTxExSteps: bfResponse.max_tx_ex_steps || '',
+      maxTxSize: bfResponse.max_tx_size,
+      maxValSize: bfResponse.max_val_size || '',
+      minFeeA: bfResponse.min_fee_a,
+      minFeeB: bfResponse.min_fee_b,
+      minPoolCost: bfResponse.min_pool_cost,
+      poolDeposit: bfResponse.pool_deposit,
+      priceMem: bfResponse.price_mem || 0,
+      priceStep: bfResponse.price_step || 0,
```

Comment:
these fallbacks technically change the behaviour. maybe consider throwing an error to prevent silent fail.